js/script.js

```diff
@@ -67,6 +67,17 @@ const createTaskCard = task => {
     return taskCard;
 };
 
+const isMatchingId = (task, taskId) => {
+    return task.id === taskId;
+};
+
+const deleteTask = taskId => {
+    const index = tasks.findIndex(task => isMatchingId(task, taskId));
+    if (index !== -1) {
+        tasks.splice(index, 1);
```

Comment:
`splice` method generally modify the original array. In this task you may not find any problem for using this method. But we should not use any method that modify our original task array. If you use `eslint` this may gives you an error. Also in different framework they will also give you an warning.

Reply:
The idea is to keep our core data store consistent. Splice might not be a bad choice when removing one item, however if you have the choice you can just rewrite your original data store with new data store.
This principle comes directly from pure functional languages like erlang, elixer where data is not mutable. So you could not mutate an array if you want. You have to create a new array with the changes and rewrite your original array. 

Thanks

Reply:
Immutability in Functional Programming Languages is a powerful concept.
Have a look at this short video where I demonstrate in Elixer, even if you remove a element
from an array, the original array remains intact.
This is because, Elixer will return a new array with the intended change and keep
the previous array same.
If you want to make changes to the original array, you have to reassign.

https://github.com/rza-wtag/Todo_App/assets/98801376/69a71cea-ea55-488e-9028-81022831229c